hack/lib/golang.sh

```diff
@@ -43,23 +41,19 @@ readonly KUBE_SUPPORTED_NODE_PLATFORMS=(
 # library is precompiled for in build/build-image/cross/Dockerfile
 readonly KUBE_SUPPORTED_CLIENT_PLATFORMS=(
```

Comment:
clients seem like a more reasonable case for targeting every architecture we can, even if we don't have a good test story. The surface area of kubelet is way smaller than API servers or kubelet... it's more plausible to me to invoke `kubectl` than run an API server or kubelet on an untested platform.

Reply:
@liggitt fair. Let's see if we get anyone 🙌🏾 and we can revert this section.

Reply:
FTR @brendanburns is in favor of keep building the clients

Reply:
+1 for keeping kubectl

We also don't have to do things like using qemu for statically linked kubectl binary, it's pretty maintainable to target lots of platforms for that binary and it seems like a win.

Reply:
Our workaround was `ssh kubectl`, which did seem like a hack

Reply:
reverted the changes to `KUBE_SUPPORTED_CLIENT_PLATFORMS`